src/sbt-test/settings/non-jvm/build.sbt

```diff
@@ -0,0 +1,7 @@
+PB.targets in Compile := Seq(
+  PB.gens.descriptorSet -> (resourceManaged in Compile).value / "newdirectory" / "descriptorset.pb",
+  PB.gens.js -> (resourceManaged in Compile).value / "js",
+  PB.gens.java -> (sourceManaged in Compile).value
+)
+
+Compile / resourceGenerators += (Compile / PB.generate).taskValue
```

Comment:
this is actually the use-case that initiated https://github.com/scalapb/protoc-bridge/pull/85: I package file descriptors to be able to serve them from my service